Refactoring/typescript

Ushlerdop opened the pull request:
Добавил typescript в приложение

Ushlerdop committed:
Installed TS packages and added the tsconfig.json

Подготовил проект под использование TS. Добавил зависимости и файл tsconfig.son

Ushlerdop committed:
Added type checking for the react-context and for all components using react-context

Добавил проверку типов в React контексте. Как при его создании, так и при объявлении Provider и Consumer

Ushlerdop committed:
Added typescript to the mobx store. Created interfaces for store

Типизировал mobx хранилище. Также добавил файл с интерфейсами для mobx в папке store.

Ushlerdop committed:
Added typescript to Forms

Типизировал TodoAddForm и TodoUpdateForm. Для последней добавил интерфейс для пропсов.

Ushlerdop committed:
Added typescript to TodoTask, TodoTaskModal and to UI/buttons/*

Добавил typescript в TodoTask, TodoTaskModal и во все кнопки в UI/buttons/*. Создал интерфейс ITaskModalProps для TodoTaskModal компонента с наследованием интерфейс ITaskObject. Последнего также теперь наследует и IUpdateFormProps.

Ushlerdop committed:
Added typescript to hocs

Добавил typescript и типы в hoc-и.

Ushlerdop committed:
Added typescript to ErrorBoundary, sleep and tasks files

Добавил typescript в ErrorBoundary компонент, а также в sleep и tasks файлы.

Ushlerdop committed:
Added typescript to LinkPage and main index. Made some scss changes and added 'go back' button to modal HOC

Добавил TypeScript в LinkPage и главный index файл. Подправил стили и добавил кнопку "назад" во все модальные окна, если сайт открыт с мобильного устройства. Убрал лишние функции в TodoTask.

Ushlerdop committed:
Added missing types and a message when the task list is empty

Добавил типы туда, где забыл их изначально указать. Также добавил сообщение, если текущий фильтр списка задач пустой.

Ushlerdop committed:
Added missing types to LanguageContext

Ushlerdop committed:
Added translation of errors when form validation fails

Добавил перевод сообщений в alert при валидации форм. Изменил методы onDescriptionChange, убрал оттуда изменение высоты textarea и перенёс это в useEffect с descriptionText в массиве зависимостей. Теперь textarea уменьшается, если добавить новую задачу с большим описанием, а не остаётся увеличенной.

Ushlerdop committed:
Added React-Hook-Form and updated Forms

Добавил React-Hook-Form и переписал формы с использованием этой библиотеки

Ushlerdop committed:
Moved validation rules for forms to a validationRules.ts file file

Переместил правила валидации в отдельный файл - validationRules.ts. Сделал надписи об ошибках красными.

Ushlerdop committed:
Added red border to inputs that failed validation

Добавил красную обводку для инпутов, которые не прошли валидацию

Ushlerdop committed:
Added a check for no changes in the TodoUpdateForm

Теперь если в окне редактирования задачи заголовок и описание не изменились, то окно для редактирования просто закрывается.

bogdankuzzmin commented on src/components/UI/buttons/DoneButton/DoneButton.tsx:
```diff
@@ -5,7 +5,7 @@ import { store } from '../../../../store';
 const cx = classNames.bind(styles);
 
-function DoneButton(props) {
+function DoneButton(props: { isDone: any; id: number }): JSX.Element {
```
isDone: boolean
У тебя где то стрелочные функции, тут function declaration. Лучше выбрать один способ и везде использовать.

bogdankuzzmin commented on src/components/UI/buttons/LanguageToggleButton/LanguageToggleButton.module.scss:
```diff
@@ -2,11 +2,17 @@
+
+@media (max-width: 992px) {
+  .container {
+    padding: 6px 6px 6px 6px;
```
```suggestion
    padding: 6px;
```

bogdankuzzmin commented on src/components/UI/buttons/LanguageToggleButton/LanguageToggleButton.module.scss:
```diff
@@ -2,11 +2,17 @@
   width: 100%;
   margin-bottom: 15px;
   padding: 10px 10px 10px 10px;
```
```suggestion
  padding: 10px;
```

bogdankuzzmin commented on src/index.css:
```diff
@@ -11,3 +13,49 @@ code {
+}
+
+::-webkit-scrollbar {
+  width: 14px; /* ширина для вертикального скролла */
+  height: 8px; /* высота для горизонтального скролла */
+  background-color: #505050;
+  border-top-right-radius: 10px;
+  border-bottom-right-radius: 10px;
+}
+/* ползунок скроллбара */
+::-webkit-scrollbar-thumb {
+  background: #8d91ca;
+  border-radius: 5px;
+}
+::-webkit-scrollbar-thumb:hover {
+  background: #7c7fa5;
+}
+/* Стрелки */
+::-webkit-scrollbar-button:vertical:start:decrement {
+  background: linear-gradient(120deg, #02141a 40%, rgba(0, 0, 0, 0) 41%),
+    linear-gradient(240deg, #02141a 40%, rgba(0, 0, 0, 0) 41%),
+    linear-gradient(0deg, #02141a 30%, rgba(0, 0, 0, 0) 31%);
+  background-color: #f6f8f4;
+}
```
комментарии тут не нужны, и так понятно что это за свойства

bogdankuzzmin commented on src/languages/interface.ts:
```diff
@@ -0,0 +1,24 @@
+  errorMessages: {
+    sameTitle: string;
+    emptyInputs: string;
+    maxLength: (maxTextInputLength: number, maxTextAreaLength: number) => string;
```
```suggestion
    maxLength(maxTextInputLength: number, maxTextAreaLength: number): string;
```

bogdankuzzmin commented on src/store/interface.ts:
```diff
@@ -0,0 +1,15 @@
+  id: number;
+}
+
+export interface IFilterStatusList {  
+  ALL: string;
+  ACTIVE: string;
+  IMPORTANT: string;
+  DONE: string;
+}
```
Можно так записать: `export type TFilterStatusList = Record<string, string>;`

Ushlerdop replied:
Я тут хочу сохранить чёткую структуру объекта. Дать понять, что внутри могут быть только ALL, ACTIVE,... т.к. других статусов фильтров не предусмотрено. Получается оставляю как есть?

Ushlerdop committed:
Made fixes based on code-review

Сделал исправления по code review

Ushlerdop committed:
Merge pull request #12 from Ushlerdop/fix-typescript

Made fixes based on code-review

Ushlerdop committed:
Merge pull request #12 from Ushlerdop/fix-typescript

Made fixes based on code-review

Ushlerdop committed:
Made fixes based on code-review

Исправил замечания из code-review

Ushlerdop committed:
Merge branch 'refactoring/typescript' into feauture/React-Hook-Form

Ushlerdop committed:
Merge pull request #11 from Ushlerdop/feauture/React-Hook-Form

Added React-Hook-Form

Ushlerdop committed:
classNames changes

Все условия по назначению классов теперь непосредственно в classNames элементов
